File: src/variables/Wrong3.cpp
#include <iostream>
#include <string>
using namespace std;

void foo(const string& x) {
  cout << "Hi!\n";
  cout << "I am " << x << "\n";
}

int main(int argc, char* argv[]) {
  if(string(argv[1]) == "trigger") {
    foo(3);
  } else {
    foo(string(argv[1]));
  }
  return 0;
}

File: src/variables/Wrong3.py
#!/usr/bin/python3

import sys

def foo(x):
    print('Hi!\n')
    print('I am ' + x)

if __name__ == "__main__" :
    if sys.argv[1] == 'trigger':
        foo(3)
    else:
        foo(sys.argv[1])
